diag_manager/fms_diag_file_object.F90

```diff
@@ -838,9 +1000,24 @@ subroutine write_axis_data(this, diag_axis)
     endif
   enddo
 
-  !TODO: closing the file here for now, just to see if it works
-  call close_file(fileobj)
 end subroutine write_axis_data
 
+!< @brief Closes the diag_file
+subroutine close_diag_file(this)
+  class(fmsDiagFileContainer_type), intent(inout), target :: this            !< The file object
+
+  !< The select types are needed here because otherwise the code will go to the
+  !! wrong close_file routine and things will not close propertly
+  select type( fileobj => this%FMS_diag_file%fileobj)
```

Comment:
You can't just call `close_file` here?  Shouldn't `close_file` have the `select type`?

Reply:
`close_file` doesn't have the `select_type`, it has interface
https://github.com/NOAA-GFDL/FMS/blob/d8208e61be2dff46cc387cbace0f1f7c2bd950fc/fms2_io/fms2_io.F90#L166-L170

If I just call close_file, the code was always going into `netcdf_file_close_wrap`, so if the fileobject was `FmsNetcdfDomainFile_t`, some stuff doesn't get deallocated 
https://github.com/NOAA-GFDL/FMS/blob/d8208e61be2dff46cc387cbace0f1f7c2bd950fc/fms2_io/fms_netcdf_domain_io.F90#L60-L66